ludwig/backend/utils/storage.py

```diff
@@ -0,0 +1,69 @@
+import contextlib
+from typing import Any, Dict, Optional, Union
+
+from ludwig.utils import data_utils
+
+CredInputs = Optional[Union[str, Dict[str, Any]]]
+
+
+DEFAULTS = "defaults"
+ARTIFACTS = "artifacts"
+DATASETS = "datasets"
+CACHE = "cache"
+
+
+class Storage:
+    def __init__(self, creds: Optional[Dict[str, Any]]):
+        self._creds = creds
+
+    @contextlib.contextmanager
+    def use(self):
```

Comment:
Maybe this should be called `use_credentials()`, i.e.

`with backend.storage.cache.use_credentials():`

Alternatively, `def use(self):` could have a doc string explaining what it does.

Reply:
I agree, changed to `use_credentials`